### PR TITLE
[12.0][FIX] mass_mailing_partner: wrong company implementation

### PR DESCRIPTION
**Pull request by chienandalu**
See issue https://github.com/OCA/social/issues/776

cc @Tecnativa

please review @ernestotejeda @ivantodorovich @pedrobaeza

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Shouldn't you create instead a parent partner from that `company_name`?

**Comment by ivantodorovich** [comment]
Or simply set the partner's company_name when it's created. According to https://github.com/odoo/odoo/commit/bf88456943bed2e4ed4a996fb4a48126e325126d it should behave as having a parent_id without actually having to create the parent. [A button is even displayed on the partner form to create it](https://github.com/odoo/odoo/blob/542cbe80ef8b71f4f48502fcc73d0c6556e679c2/odoo/addons/base/views/res_partner_views.xml#L151)

**Comment by chienandalu** [comment]
Yes, I'll go for @ivantodorovich take.

File: mass_mailing_partner/models/mail_mass_mailing_contact.py [hunk 1]
```diff
@@ -96,7 +86,7 @@ def _prepare_partner(self):
             'email': self.email,
             'country_id': self.country_id.id,
             'title': self.title_id.id,
-            'company_id': self._get_company(),
+            'company_id': self.company_name,
```
Comment (pedrobaeza) [inline mass_mailing_partner/models/mail_mass_mailing_contact.py]:
```suggestion
            'company_name': self.company_name,
```

**Comment by pedrobaeza** [comment]
About `company_id` field, is mailing contact linked to a company_id? If so, we should respect it.

File: mass_mailing_partner/models/mail_mass_mailing_contact.py (hunk 1 above)
Comment (chienandalu) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: oops, yes :sweat_smile:

**Comment by chienandalu** [comment]
Well, all the mass_mailing stuff is quite company agnostic

**Comment by pedrobaeza** [comment]
Then I think we should force `company_id=False` in the contact for avoiding problems.

**Comment by chienandalu** [comment]
Ok, done

**Review by pedrobaeza — APPROVED** [review]

**Review by ivantodorovich — APPROVED** [review]

**Pull request merged**